chore: readme update (#64)
rudderlabs/rudder-typer: README.md

### RudderStack JavaScript SDK

There are two ways to get started with Rudder-Typer in your browser:

> #### 1. Using CDN (Snippet)

Reviewer: Can we also point in the instructions at the end to the appropriate sample application for more details?

Reviewer: I felt I didn't have to point this out. Let's do the same for NPM.

Author: Done

- Paste the snippet from Rudderstack dashboard to your HTML file. [Ref](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/quickstart/#using-cdn)
- If you are using typescript add `@rudderstack/analytics-js` as dev dependency.

```sh
npm install --save-dev  @rudderstack/analytics-js
```

- Execute the following command to generate a bundle from the RudderTyper client:

```sh
npx browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
```

> - For typescript project add `tsify` npm package as dev dependency and run the following command:

Reviewer: ```suggestion
- For TypeScript analytics client, add `tsify` NPM package as a dev dependency and run the following command to generate the bundle:
```

```sh

> npx browserify analytics/index.js -p [ tsify ] --standalone rudderTyper >  rudderTyperBundle.js

Reviewer: This is the main change in the command from the previous instruction. Please be a little more cautious.
```suggestion
npx browserify analytics/index.ts -p [ tsify ] --standalone rudderTyper >  rudderTyperBundle.js
```

```

- Now import your RudderTyper client and send events

```html

> <script src="./rudderTyperBundle.js"></script>

Reviewer: What should be the order of the installation of the SDK snippet and this bundle?

Can you clarify this with an example without using the actual snippet?

### RudderStack JavaScript SDK

> There are two ways to get started with Rudder-Typer in your browser:

Reviewer 2: ```suggestion
There are two ways to get started with RudderTyper in your browser:
```

> #### 1. Using CDN (Snippet)

Reviewer 2: ```suggestion
#### 1. Using the JavaScript SDK snippet
```

> - Paste the snippet from the Rudderstack dashboard to your HTML file. [Ref](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/quickstart/#using-cdn)

Reviewer 2: ```suggestion
- Paste the [JavaScript SDK snippet](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/quickstart/#using-cdn) from the RudderStack dashboard to your HTML file.
```

- If you use TypeScript, add `@rudderstack/analytics-js` as a dev dependency.

```sh
npm install --save-dev  @rudderstack/analytics-js
```

> - Execute the following command to generate a bundle from the RudderTyper client:

Reviewer 2: ```suggestion
- Run the following command to generate a bundle from the RudderTyper client:
```

```sh
npx browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
```

> - For TypeScript analytics client, add `tsify` NPM package as a dev dependency and run the following command to generate the bundle:

Reviewer 2: ```suggestion
- For the TypeScript analytics client, add the NPM package `tsify` as a dependency and run the following command to generate the bundle:
```

```sh
npx browserify analytics/index.ts -p [ tsify ] --standalone rudderTyper >  rudderTyperBundle.js
```

> - Now import your RudderTyper client and send events

Reviewer 2: ```suggestion
- Import your RudderTyper client and send events, as shown:
```

- Execute the following command to generate a bundle from the RudderTyper client:

```sh
npx browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
```

For typescript project add `tsify` npm package as dev dependency and run the following command:

```sh
npx browserify analytics/index.js -p [ tsify ] --standalone rudderTyper >  rudderTyperBundle.js
```

- Now you can make calls from your `html` file as shown:

> ```html

Reviewer: Let's not add the RudderStack snippet here. Please update the instructions to point to the specific docs section.

Author: Addressed

</script>
```

#### 2. Using NPM

Import the RudderTyper-generated client and make the calls if your framework supports them.

```javascript
// Import RudderStack JS SDK and initialize it
import { RudderAnalytics } from '@rudderstack/analytics-js';
// Import your auto-generated RudderTyper client:
import { RudderTyperAnalytics } from './analytics/index';

const rudderAnalytics = new RudderAnalytics();
rudderAnalytics.load(WRITE_KEY, DATA_PLANE_URL, {});

// Pass in your @rudderstack/analytics-js instance to RudderTyper client
RudderTyperAnalytics.setRudderTyperOptions({
  analytics: rudderAnalytics,
});

// Issue your first RudderTyper track call!
RudderTyperAnalytics.orderCompleted({
  orderID: 'ck-f306fe0e-cc21-445a-9caa-08245a9aa52c',
  total: 39.99,
});
```

> > Note: Remember to replace all instances of `WRITE_KEY` and `DATA_PLANE_URL` in the snippet with your write key and data plane URL values respectively.

Reviewer: Please move this to the CDN section.

Reviewer: If they use the snippet from the RS dashboard, these values will be prefilled. So, no need for this instruction itself. Apologies if your original intention was the same.

If you think it'll be helpful for the NPM case, keep it but phrase it appropriately.

Author: Done

</script>
```

> > Check out the [sample application](https://github.com/rudderlabs/rudder-typer/tree/develop/examples/js-cdn-typescript) for reference.

Reviewer 2: ```suggestion
> See the [sample application](https://github.com/rudderlabs/rudder-typer/tree/develop/examples/js-cdn-typescript) for reference.
```

#### 2. Using NPM

Import the RudderTyper-generated client and make the calls if your framework supports them.

```javascript
// Import RudderStack JS SDK and initialize it
import { RudderAnalytics } from '@rudderstack/analytics-js';
// Import your auto-generated RudderTyper client:
import { RudderTyperAnalytics } from './analytics/index';

const rudderAnalytics = new RudderAnalytics();
rudderAnalytics.load(WRITE_KEY, DATA_PLANE_URL, {});

// Pass in your @rudderstack/analytics-js instance to RudderTyper client
RudderTyperAnalytics.setRudderTyperOptions({
  analytics: rudderAnalytics,
});

// Issue your first RudderTyper track call!
RudderTyperAnalytics.orderCompleted({
  orderID: 'ck-f306fe0e-cc21-445a-9caa-08245a9aa52c',
  total: 39.99,
});
```

> > Note: Remember to replace all instances of `WRITE_KEY` and `DATA_PLANE_URL` with your write key and data plane URL values respectively.

Reviewer 2: ```suggestion
Note the following: 

- Remember to replace `WRITE_KEY` and `DATA_PLANE_URL` in the above snippet with the write key of your JavaScript source and the data plane URL respectively.
- Run `npx rudder-typer` to regenerate your RudderTyper client every time you update your tracking plan.
```

> 

Reviewer 2: ```suggestion
```

> > Note: Run `npx rudder-typer` to regenerate your RudderTyper client. You need to do this each time you update your Tracking Plan.

Reviewer 2: ```suggestion
```

> > Check out the [sample application](https://github.com/rudderlabs/rudder-typer/tree/develop/examples/js-npm-typescript) for reference.

Reviewer 2: ```suggestion
See the [sample application](https://github.com/rudderlabs/rudder-typer/tree/develop/examples/js-npm-typescript) for reference.
```

</script>
```

> > Note: Remember to replace all `WRITE_KEY` and `DATA_PLANE_URL` with your write key and data plane url.

Reviewer: ```suggestion
> Note: Remember to replace all instances of `WRITE_KEY` and `DATA_PLANE_URL` in the snippet with your write key and data plane URL values respectively.
```

Author: Done
